src/Common/src/System/Text/DBCSDecoder.cs

```diff
@@ -0,0 +1,293 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+using System.Diagnostics;
+
+namespace System.Text
+{
+    internal class DecoderDBCS : Decoder
```

Comment:
Decoder is serializable, but this new DecoderDBCS is not marked [Serializable].  Is there any situation where the Decoder instance previously used in whatever situation this is being used now could have been handed out to user code and serialized?